src/Lean/Parser/Command.lean

```diff
@@ -754,6 +754,30 @@ list, so it should be brief.
   optional (docComment >> ppLine) >>
   "tactic_extension " >> ident
 
+/--
+Document a recommended spelling for a notation in identifiers.
+
+```
+/-- some additional info -/
+recommended_spelling "∧" "and" And «term_∧_»
+```
+
+will do the following:
+* Adds the sentence "The recommended spelling of `∧` in identifier is `and` (some additional info)."
+  to the end of the docstring for `And` and for `∧`. If the additional info is more than a single
+  line, it will be placed below the sentence instead of in parentheses.
+* Registers this information in an environment extension, so that it will later be possible to
+  generate a table with all recommended spellings.
+
+You can add attach the recommended spelling to as many declarations as you want. It is recommended
+to attach the recommended spelling to all relevant parsers as well as the declaration the parsers
```

Comment:
How should users know what the name of the parser is if it's auto-generated?

Reply:
Batteries has `#help term` for that, but that's not really something we can put into this documentation string I'm afraid...

Reply:
There are a few ways to do it that we could indicate here:
 1. Hovering the declaration
 2. Using `(name := ...)`

I prefer recommending number 2 to users, because auto-generated names can in principle change due to conflicts in transitive imports.

All this challenge goes away, of course, if we get a mapping between syntax and the name that it represents, like we talked about elsewhere.